data-prepper-plugins/rss-source/src/main/java/org/opensearch/dataprepper/plugins/source/rss/RSSSource.java

```diff
@@ -24,20 +33,47 @@ public class RSSSource implements Source {
     private final RSSSourceConfig rssSourceConfig;
 
     @DataPrepperPluginConstructor
-    public RSSSource(PluginMetrics pluginMetrics, final RSSSourceConfig rssSourceConfig) {
+    public RSSSource(final PluginMetrics pluginMetrics, final RSSSourceConfig rssSourceConfig) {
         this.pluginMetrics = pluginMetrics;
         this.rssSourceConfig = rssSourceConfig;
     }
 
-
-
     @Override
     public void start(Buffer buffer) {
         // TODO: Add to buffer
+        if (buffer == null) {
+            throw new IllegalStateException("Buffer is null");
+        }
     }
 
     @Override
     public void stop() {
         // TODO: Stop the service
     }
+
+    void extractItemsFromRssFeed(final RSSSourceConfig rssSourceConfig) {
```

Comment:
1. The user will provide the `url` in their Pipeline configuration and it will be picked up during startup.
2. The entire feed will be read at intervals, known as `pollingFrequency`, for which default is 5 minutes. The logic will be added as a follow-up PR.